lldb/source/Target/Platform.cpp

```diff
@@ -989,6 +989,14 @@ uint32_t Platform::FindProcesses(const ProcessInstanceInfoMatch &match_info,
   return match_count;
 }
 
+ProcessInstanceInfoList Platform::GetAllProcesses() {
+  ProcessInstanceInfoList processes;
+  ProcessInstanceInfoMatch match;
+  assert(match.MatchAllProcesses());
```

Comment:
why do we need this assert here ?

Reply:
This ensures that the default constructed `ProcessInstanceInfoMatch` actually matches all processes. This is the method that the implementation keys off of, so it seemed appropriate to have sanity check here.

Reply:
Sounds reasonable.